plugins/generic/pdfJsViewer/templates/display.tpl

```diff
@@ -48,10 +48,11 @@
 
 	<script type="text/javascript" src="{$pluginUrl}/pdf.js/build/pdf.js"></script>
 	<script type="text/javascript">
+		var pluginUrl = {$pluginUrl|json_encode};
 		{literal}
 			$(document).ready(function() {
-				PDFJS.workerSrc='{/literal}{$pluginUrl}/pdf.js/build/pdf.worker.js{literal}';
-				PDFJS.getDocument({/literal}'{$downloadUrl|escape:"javascript"}'{literal}).then(function(pdf) {
+				pdfjsLib.GlobalWorkerOptions.workerSrc = pluginUrl + '/pdf.js/build/pdf.worker.js';
+				pdfjsLib.getDocument({/literal}'{$downloadUrl|escape:"javascript"}'{literal}).promise.then(function(pdf) {
```

Comment:
Promises are not supported in all the browsers that we target. Did you check whether this is the native `Promise` implementation being invoked or if pdfjsLib has polyfilled it?